### DISCUSSION TITLE
Trying to build on macOS

### DISCUSSION THREAD

**Discussion by jenstroeger** [i3/i3#5121] [Q&A] [answered]
Hello,

I’m trying to build `i3wm` from source on macOS 10.14 because there’s [no Brew package](https://github.com/orgs/Homebrew/discussions/3627) and the [MacPorts package is broken](https://trac.macports.org/ticket/65811).

Trying to run through `meson ..` I’m now stuck with
```
../meson.build:325:0: ERROR: C shared or static library 'ev' not found
```
although I’ve got two event libraries installed:
```
libev @4.33_1 (active)
libevent @2.1.12_1 (active)
```
I also noticed
```
Library rt found: NO
```
and wonder which library that would be? (I’m not too familiar with debugging the [`meson` build system](https://mesonbuild.com/)).

Any suggestions would be welcome! Thank you

**Comment by Airblader** [comment]
This is off topic, but what's the point of building i3 on MacOS?

**Comment by jenstroeger** [comment]
So I can open [XQuartz](https://www.xquartz.org/) on one single “Spaces” desktop and then have a tiling wm with its own [workspaces](https://i3wm.org/docs/userguide.html#_using_workspaces) on that one desktop.

Been working that way for years and it’s perfect. But now breaking changes and package neglect force me to build `i3wm` myself.

**Reply by Airblader** [reply]
I see, thanks. I won't be of any help with the issue, unfortunately.

**Comment by jenstroeger** [comment]
@Airblader can you recommend a more suitable place to ask this question? Does the `i3` community have a developer list or some such?

**Reply by Airblader** [reply]
The most active place community-wise is probably the i3wm subreddit. There is also a mailing list and an IRC channel (no idea how active that one is, I never joined it). You'll find those things on the website.

**Comment by vincentbernat** [comment]
On Linux, librt is part of the GNU Libc since quite some time. I suppose that's also the case on macOS, so no worry. As for libev, no clue. Shouldn't you have development headers as well? Search for `ev.h` for example.

**Reply by jenstroeger** [reply]
> As for libev, no clue. Shouldn't you have development headers as well? Search for `ev.h` for example.

Oddly enough, the headers and libs exist but it looks like the compiler can’t find them:
```
Running compile:
Working directory:  /Users/savage/opt/src/i3/build/meson-private/tmp4q_33hpg
Command line:  cc /Users/savage/opt/src/i3/build/meson-private/tmp4q_33hpg/testfile.c -o /Users/savage/opt/src/i3/build/meson-private/tmp4q_33hpg/output.exe -O0 -Werror=implicit-function-declaration -lev -Wl,-undefined,dynamic_lookup 

Code:
 int main(void) { return 0; }

Compiler stdout:
 
Compiler stderr:
 ld: library not found for -lev
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```
Running the command manually with the same test source:
```
cc bla.c -o bla -O0 -Werror=implicit-function-declaration -lev -Wl,-undefined,dynamic_lookup -L /opt/local/lib
```
works—notice the `-L` at the end. Can I tell `meson` to search additional paths, or should I add symlinks into the system paths (which I’d like to avoid).

**Comment by jenstroeger** [comment, marked as answer]
Alright. After [some fiddling](https://trac.macports.org/ticket/65811#comment:4) I was able to build `i3wm` using MacPorts. However, now I need to sort out the missing `i3bar` but perhaps that’s a separate discussion…

**Reply by m0ne** [reply]
do you want to elaborate? interested in how you've done it and how it is going.

**Reply by jenstroeger** [reply]
I never followed up with  building `i3bar` but did the fiddling link not work?

**Reply by m0ne** [reply]
I haven't had time to try this yet, but thanks for the feedback
